### DISCUSSION TITLE
Closing the floating via the `closeOnFocusOut` doesn't provide a `reason` in the `onOpenChange` callback

### DISCUSSION THREAD

**Discussion by oleksandr-danylchenko** [floating-ui/floating-ui#3012] [Ideas]
## Context
In https://github.com/floating-ui/floating-ui/pull/2615 the `reason` argument was added to the `onOpenChange` callback. It provides opening/closing reasons for most of the lifecycle events.

## Issue

However, there's no `reason` provided for the closing event caused by losing focus on the floating element. Such behavior gets enabled when you add the [`closeOnFocusOut`](https://floating-ui.com/docs/floatingfocusmanager#closeonfocusout) property on the `FloatingFocusManager`.

## Example
Codesandbox - https://codesandbox.io/p/sandbox/fragrant-rain-wky3sm

https://github.com/user-attachments/assets/f4e7e796-169e-4610-805c-2facc4b11976

**Comment by atomiks** [comment]
To be honest I didn't really know what to call this so I just left it out (I also thought it might have been too low-level to add in the first place). The docs do say not every open change event has a reason attached (I think this one has the event though? so you could check that.)

**Reply by oleksandr-danylchenko** [reply]
Yep, the `onOpenChange` **does provide** the `Focus` event when the popup closes. However, checking it doesn't feel as robust as having an explicit `reason` (which is super useful for the `esc` handling though 🤩).

**Reply by oleksandr-danylchenko** [reply]
UPD: I provided the reason emission in https://github.com/floating-ui/floating-ui/pull/3015

**Comment by oleksandr-danylchenko** [comment]
The `focus-out` reason got exposed and added into the [@floating-ui/react@0.26.23](https://github.com/floating-ui/floating-ui/releases/tag/%40floating-ui%2Freact%400.26.23) release  👌🏻
